sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/models/FeatureFlagConfigurationSetting.java

```diff
@@ -136,7 +136,11 @@ public String getValue() {
      */
     @Override
     public FeatureFlagConfigurationSetting setKey(String key) {
-        super.setKey(key);
+        if (key != null && key.startsWith(FeatureFlagConfigurationSetting.KEY_PREFIX)) {
```

Comment:
Few more thoughts:

- Should a null `key` throw an error?
- Instead of substring when the `key` starts with `KEY_PREFIX` we should instead add the `KEY_PREFIX` if it doesn't start with the `KEY_PREFIX`.

```java
if (key != null && !key.startsWith(KEY_PREFIX)) {
    super.setKey(KEY_PREFIX + key);
} else {
    super.setKey(key);
}

return this;
```

Reply:
`key` is required and cannot be null. But we don't do client-side validation